.github/workflows/publish-crates.yml

```diff
@@ -0,0 +1,80 @@
+name: Publish Crates
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to publish (without v prefix)'
+        required: true
+        type: string
+
+permissions:
+  contents: read
+
+jobs:
+  publish-crates:
+    name: Publish crates to crates.io
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      # Update versions in Cargo.toml files
+      - name: Update crate versions
+        run: |
+          VERSION=${{ inputs.version }}
+          # Update version in each Cargo.toml
+          for crate in common crypto serde class-hash; do
+            # Update the package version
+            sed -i "s/^version = \".*\"/version = \"$VERSION\"/" "crates/$crate/Cargo.toml"
+            # Also update any internal dependencies to use the same version
+            sed -i "s/{ path = \"..\/common\" }/{ version = \"$VERSION\" }/" "crates/$crate/Cargo.toml"
+            sed -i "s/{ path = \"..\/crypto\" }/{ version = \"$VERSION\" }/" "crates/$crate/Cargo.toml"
+            sed -i "s/{ path = \"..\/serde\" }/{ version = \"$VERSION\" }/" "crates/$crate/Cargo.toml"
+          done
+
+      - uses: dtolnay/rust-toolchain@stable
+
+      # Verify all crates build and test before publishing
+      - name: Verify crates
+        run: |
+          cargo test -p pathfinder-crypto
+          cargo test -p pathfinder-common
+          cargo test -p pathfinder-serde
+          cargo test -p pathfinder-class-hash
```

Comment:
You can specify multiple packages for a single `cargo test` invocation:

```
cargo test -p pathfinder-crypto -p pathfinder-common -p pathfinder-serde -p pathfinder-class-hash
```

In addition I'd add at least a `timeout 5m` to each of these just to make sure we avoid issues.

We should also test if `cargo doc` works properly.